panel/chat/feed.py

```diff
@@ -335,12 +335,17 @@ def _upsert_message(
         if isinstance(value, dict):
             user = value.get("user", user)
             avatar = value.get("avatar")
+
         if message is not None:
```

Comment:
Could there be a situation where `message is not None and isinstance(value, ChatMessage)`? And how should that be handled?

Reply:
ChatMessage overrides it I think in

        elif isinstance(value, ChatMessage):
            # ChatMessage is not created yet, but a ChatMessage is passed; use it
            self._replace_placeholder(value)
            return value

Reply:
I can't see that. Where do you see that?

![image](https://github.com/holoviz/panel/assets/42288570/a0e51cd2-0ce5-4094-8162-c903399b99bd)

Reply:
:) Ok, perhaps not ChatMessage, but dict works as expected.

Reply:
Should you add something like?

```python
...
elif isinstance(value, ChatMessage):
    user = value.user
    avatar = value.avatar
    value = value.value
```

Reply:
No, we don't need to extract the values if it's a ChatMessage and message is None.

It does look neater though.